simpleGetWithHostnameVerificationEnabledTest fails on macOS, because 127.0.0.2 is not bindable

**Jetty version(s)**
Jetty 12.0.x

**Jetty Environment**
core

**Java version/vendor** `(use: java -version)`
openjdk version "21.0.8" 2025-07-15
OpenJDK Runtime Environment Homebrew (build 21.0.8)
OpenJDK 64-Bit Server VM Homebrew (build 21.0.8, mixed mode, sharing)

**OS type/version**
macOS 15.5 (24F74)

**Description**
Hello, the simpleGetWithHostnameVerificationEnabledTest test fails, when I run "mvn install" on my Macbook Air M1 with Homebrew OpenJDK 21, because (differently from Linux) the address 127.0.0.2 is not bindable (without an extra manual setup like `sudo ifconfig lo0 alias 127.0.0.2 up`):

```
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 5.873 s <<< FAILURE! -- in org.eclipse.jetty.client.HostnameVerificationTest
[ERROR] org.eclipse.jetty.client.HostnameVerificationTest.simpleGetWithHostnameVerificationEnabledTest -- Time elapsed: 5.823 s <<< FAILURE!
java.lang.AssertionError: 

Expected: an instance of javax.net.ssl.SSLHandshakeException
     but: <java.net.SocketTimeoutException: Connect Timeout> is a java.net.SocketTimeoutException
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
        at org.eclipse.jetty.client@12.0.25-SNAPSHOT/org.eclipse.jetty.client.HostnameVerificationTest.simpleGetWithHostnameVerificationEnabledTest(HostnameVerificationTest.java:129)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   HostnameVerificationTest.simpleGetWithHostnameVerificationEnabledTest:129 
Expected: an instance of javax.net.ssl.SSLHandshakeException
     but: <java.net.SocketTimeoutException: Connect Timeout> is a java.net.SocketTimeoutException
```

**How to reproduce?**

```
cd jetty-core/jetty-client && mvn clean test -Dtest=HostnameVerificationTest#simpleGetWithHostnameVerificationEnabledTest -Dmaven.build.cache.enabled=false
```

Here are the logs (please wait for my upcoming PR):

[simpleGetWithHostnameVerificationEnabledTest-fails.txt](https://github.com/user-attachments/files/21706242/simpleGetWithHostnameVerificationEnabledTest-fails.txt)
[simpleGetWithHostnameVerificationEnabledTest-fixed.txt](https://github.com/user-attachments/files/21706241/simpleGetWithHostnameVerificationEnabledTest-fixed.txt)

--- a/jetty-core/jetty-client/src/test/java/org/eclipse/jetty/client/HostnameVerificationTest.java
+++ b/jetty-core/jetty-client/src/test/java/org/eclipse/jetty/client/HostnameVerificationTest.java
@@ -13,6 +13,9 @@
 
 package org.eclipse.jetty.client;
 
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
 import java.security.cert.CertificateException;
 import java.util.concurrent.ExecutionException;
 import javax.net.ssl.SSLHandshakeException;
@@ -36,6 +39,7 @@ import org.eclipse.jetty.util.ssl.SslContextFactory;
 import org.eclipse.jetty.util.thread.QueuedThreadPool;
 import org.hamcrest.Matchers;
 import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assumptions;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
@@ -121,8 +125,21 @@ public class HostnameVerificationTest
     @Test
     public void simpleGetWithHostnameVerificationEnabledTest()
     {
+        String bindAddress = "127.0.0.2";
+
+        // Pre-check if 127.0.0.2 is bindable (on macOS probably not)
+        try (ServerSocket testSocket = new ServerSocket())
+        {
+            testSocket.bind(new InetSocketAddress(bindAddress, 0));
+        }
+        catch (IOException e)
+        {
+            Assumptions.assumeTrue(false,
+                "Cannot bind to " + bindAddress + " address on this system");
+        }
+
         clientSslContextFactory.setEndpointIdentificationAlgorithm("HTTPS");
-        String uri = "https://127.0.0.2:" + connector.getLocalPort() + "/";
+        String uri = "https://" + bindAddress + ":" + connector.getLocalPort() + "/";
 
         ExecutionException x = assertThrows(ExecutionException.class, () -> client.GET(uri));
         Throwable cause = x.getCause();

